format_spec.md

```diff
@@ -0,0 +1,339 @@
+
+---
+
+### `minecraftVersions`
+
+This field MUST be an array listing compatible Minecraft versions for this Project version.
+
+---
+
+### `dependencies` (optional)
+
+This field MUST be an array containing information about dependencies that may or may not be required for this project. Each object in this array contains the following fields:
+
+##### `type`
+
+The type of dependency this is. Valid options are:
+
+- `asset` - For example, a required library mod
+- `framework` - For example, a required modloader
+
+Frameworks exist as a separate type because they may require a more advanced installation process.
+
+##### `src`
+
+This field contains where this dependency can be downloaded. It contains the name of the host, in an all-lowercase string. Examples include: `curseforge`, `diluv` and `modrinth`. If the `type` field is set to `framework`, this field CANNOT be present.
+
+##### `id`
+
+The ID of the required dependency. If the dependency is a required asset and not a framework, then the ID of the asset stored at the host specified in the `src` field should be used. If the dependency is a required framework, then a framework from the list of standardized framework IDs should be used:
+
+- `fabric-loader`
+- `forge`
+- `liteloader`
+- `rift`
+
+##### `required`
+
+This field MUST be present on dependencies with the `asset` type. It CANNOT be present on dependencies with the `framework` type. This field is a boolean that indicates whether a dependency is required in order for the Project requiring it to function properly.
+
+##### `version`
+
+This field MUST be either an array or String. If this field is an array, it MUST contain a list of compatible versions. If this field is a String, it MUST contain a Semantic Versioning comparison String. An example of a comparison string is `>= 25.0.219`
+
+---
+
+### `files`
+
+This field contains the downloadable files of a Version. It MUST be an array of File Objects.
+File Objects contain information on a downloadable file. Fields are as follows:
+
+##### `name`
+
+There is no restriction on what this value can be. However, it is RECOMMEND that this be the name of the file. Launchers and other automation tools MAY ignore the value. This value will be shown to users.
+
+##### `sha256`
+
+An SHA256 checksum of the file. This may be used by launchers in order to verify download integrity.
+
+##### `rel` (optional)
+
+The relation of this file. This MUST be one of the following values:
+
+- `primary` - The primary file of this version. This is what launchers should download.
+- `source` - Source code of this version
+- `deobf` - Deobfuscation file for this version
+
+If this field is not present, the file should be assumed to be `primary`. The values of this field CAN be present in multiple files. For example, a version can have a file for both the Forge and Fabric modloaders, and both of those files can have the `rel` field set to `primary`.
```

Comment:
Why should that be assumed? Is it just in case the index team is too lazy to add it?